### DISCUSSION TITLE
caps_lock remap

### DISCUSSION THREAD

**Discussion by nandv** [sebanc/brunch#1149] [General]
This is a chromeOS newbie question so bear with me. 
I am really enjoying the newly revived thinkpad yoga 12 on ChromeOS, everything works great, including tablet mode, and particularly the stylus, with palm-rejection!

Except for one thing: in Windows and Linux, I can easily remap the caps_lock (right below Tab key) to left Ctrl, but I have a really hard time doing it in ChromeOS. I can do it in Cloudready but there's no Android support so ..
Any suggestion?

**Comment by obzbdc** [comment]
You should be able to go to Settings > Device > Keyboard and change it there:
![image](https://user-images.githubusercontent.com/53860530/125787926-00993357-9003-406a-9b7d-7159ce605725.png)

**Comment by nandv** [comment]
Please advise which 'board'/version of ChromeOS you're using?
Mine is the recommended 'rammus' and there's no 'caps_lock' remap in the keyboard setting:
![Screenshot 2021-07-15 21 40 41](https://user-images.githubusercontent.com/16523410/125799718-5e51c9d0-8551-4bb2-bb24-5e5d8215d389.png)

**Comment by obzbdc** [comment]
Hmmm, I'm running Rammus on a Dell Latitude 7290 and have the option to remap the Caps Lock key but don't have it on a Dell XPS 13 2017 running Eve.

Edit: I've just changed the XPS over to Rammus and still don't have the option so I don't think it's build related.

**Reply by nandv** [reply]
Looks like it's time to try some older boards. I will update here if progress is made. 
Thanks for the notes.

**Reply by nandv** [reply]
I've tried more than 10 ROMs without ANY luck. 
Then one day I thought, why not connect to an external BT keyboard. Sure enough, the keyboard remap setting, 'caps lock' shows up. Then I swapped the caps_lock with ctrl. Disconnect the BT keyboard, and the remap still works for the internal keyboard!! 
What's even better is the config is persistent across reboots!! 
This problem is solved!!

**Reply by obzbdc** [reply]
It didn't even occur to me that this would be a factor, the 7290 where I had the option showing has an external keyboard connected too so that may well be the factor.

**Comment by Hiwi72** [comment]
Hi nandv,

sorry for digging out this old post.
but i am trying to get stylus WITH palmrejection to work on my yoga S1 (very similar to yours i think) How did u get it working, any ideas?
Because my stylus works great and is recognized but as soon as i have my palm resting in the touchscreen stuff is not working anymore.

thanks!
